src/repositories/request-repository.ts

```diff
@@ -37,327 +81,256 @@ const countRetryMetrics = (requests: Request[], anchoringDeadline: Date): void =
 }
 
 @singleton()
-@EntityRepository(Request)
-export class RequestRepository extends Repository<Request> {
-  private readonly isolationLevel
-
-  constructor(
-    @inject('config') private config?: Config,
-    @inject('dbConnection') private connection?: Connection
-  ) {
-    super()
-    this.isolationLevel =
-      this.connection.options.type === 'sqlite' ? 'SERIALIZABLE' : 'REPEATABLE READ'
-  }
+export class RequestRepository {
+  constructor(@inject('dbConnection') private connection?: Knex) {}
 
   /**
    * Create/updates client request
    * @param request - Request
+   * @param options
+   * @returns A promise that resolves to the created request
+   */
+  public async createOrUpdate(request: Request, options: Options = {}): Promise<Request> {
+    const { connection = this.connection } = options
+    const [{ id }] = await connection
+      .table(TABLE_NAME)
+      .insert(request, ['id'])
+      .onConflict('cid')
+      .merge()
+
+    const created = await connection.table(TABLE_NAME).first().where({ id })
+
+    logEvent.db({
+      type: 'request',
+      action: 'upsert',
+      ...request,
+      createdAt: created.createdAt.getTime(),
+      updatedAt: created.updatedAt.getTime(),
+    })
+
+    return created
+  }
+
+  /**
```

Comment:
sorry, this file is really hard to review with so much code motion.  Can you at least put the functions back into the same relative order to each other so it's easier to compare functions before and after?

Reply:
Yup will do and thanks for all the feedback. Its very helpful. I'll take a look at the PR and try to make it easier to review and remove refactors that are not necessary. I also have fresh eyes now which will help a lot in figuring out where I went too hard.